substrate/client/network/src/discovery.rs

```diff
@@ -341,6 +349,15 @@ impl DiscoveryBehaviour {
 			k.add_address(&peer_id, addr.clone());
 		}
 
+		match self.aux_address_store.get(&peer_id) {
+			None => {
+				self.aux_address_store.insert(peer_id, HashSet::from_iter([addr.clone()]));
+			},
+			Some(addresses) => {
+				addresses.insert(addr.clone());
+			},
+		}
+
```

Comment:
I think there may be a bug in the code right now. Peers discovered using Kademlia random walks should be added to the routing tables but that doesn't seem to be happening right now: https://github.com/paritytech/polkadot-sdk/blob/771c3fbde7ba3849298d6fc7df390fc6b5a04d2f/substrate/client/network/src/service.rs#L1465-L1467

https://github.com/paritytech/polkadot-sdk/blob/771c3fbde7ba3849298d6fc7df390fc6b5a04d2f/substrate/client/network/src/discovery.rs#L459-L464

Reply:
> don't know if such addresses are also removed from DHT

`libp2p-kad` (i.e. the DHT) only replaces an offline node iff it has 20 nodes for the given Kademlia key AND it just discovered an online node that it can replace the offline node with.

See below for details.

https://github.com/libp2p/rust-libp2p/blob/91fb6a16ebad5c3d6bdf29c2c688938a5855df78/protocols/kad/src/behaviour.rs#L1875-L1887

> Peers discovered using Kademlia random walks should be added to the routing tables but that doesn't seem to be happening right now: 

Peers discovered using Kademlia's random walks are added to routing table automatically unless you set [`KademliaBucketInserts::Manual`](https://github.com/libp2p/rust-libp2p/blob/91fb6a16ebad5c3d6bdf29c2c688938a5855df78/protocols/kad/src/behaviour.rs#L124-L146).

Reply:
We specify `KademliaBucketInserts::Manual` [here](https://github.com/paritytech/polkadot-sdk/blob/945ebbbcf66646be13d5b1d1bc26c8b0d3296d9e/substrate/client/network/src/discovery.rs#L221). For some reason the `Discovered` event doesn't contain the addresses that we should add to the DHT so I'm not sure what has happened here. Either the code is broken and still somehow works or the comments are outdated.